### PR TITLE
Upgrade to Laravel 11

### PR DESCRIPTION
**Pull request by mateusjunges**
This PR upgrades the app to Laravel 11.

We have to update the .env keys before deploying this.

### CODE REVIEW

File: database/migrations/2024_08_09_172027_create_cache_table.php
Comment (jakebathman) [inline database/migrations/2024_08_09_172027_create_cache_table.php:1]: What are these tables used for? I couldn't find references in the code to these tables, so maybe you're in the middle of something.

File: app/Models/User.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 use Illuminate\Foundation\Auth\User as Authenticatable;
 use Illuminate\Notifications\Notifiable;
 
-class User extends Authenticatable
+final class User extends Authenticatable
```
Comment (jakebathman) [inline app/Models/User.php:9]: As far as I know, we usually shy away from using final for any classes unless we _really_ need to for some reason.

```suggestion
class User extends Authenticatable
```

File: app/Providers/RouteServiceProvider.php [hunk 2]
```diff
@@ -8,7 +8,7 @@
 use Illuminate\Support\Facades\RateLimiter;
 use Illuminate\Support\Facades\Route;
 
-class RouteServiceProvider extends ServiceProvider
+final class RouteServiceProvider extends ServiceProvider
```
Comment (jakebathman) [inline app/Providers/RouteServiceProvider.php:11]:
```suggestion
class RouteServiceProvider extends ServiceProvider
```

**Review by jakebathman — CHANGES_REQUESTED** [review]
Looks good, thanks for your hard work on this. Just a few changes and one question, but I've tested it all locally and it's otherwise good to go. I can help with the Flare key on prod as well once this is merged.

File: database/migrations/2024_08_09_172027_create_cache_table.php
Comment (mateusjunges) [reply database/migrations/2024_08_09_172027_create_cache_table.php:1]: I forgot to change the cache env key from `CACHE_DRIVER` to `CACHE_STORE`, and `database` is the default cache driver in L11, that's why I published the migration. Updated my `.env` and reverted that commit.

**Review by jakebathman — APPROVED** [review]

**Pull request merged**